pyanaconda/ui/gui/spokes/installation_source.py

```diff
@@ -427,15 +429,22 @@ def __init__(self, *args, **kwargs):
         self._repo_store_lock = threading.Lock()
 
         self._network_module = NETWORK.get_proxy()
+        self._subscription_module = SUBSCRIPTION.get_proxy()
         self._device_tree = STORAGE.get_proxy(DEVICE_TREE)
 
         self._treeinfo_repos_already_disabled = False
 
     def apply(self):
         source_changed = self._update_payload_source()
         repo_changed = self._update_payload_repos()
-
-        if source_changed or repo_changed or self._error:
+        cdn_source = check_cdn_is_installation_source(self.payload)
+        subscribed = self._subscription_module.IsSubscriptionAttached
+        # If CDN is the current installation source but no subscription is
+        # attached there is no need to refresh the installation source,
+        # as without the subscription tokens the refresh would fail anyway.
+        if cdn_source and not subscribed:
+            log.debug("CDN source but no subscribtion attached - skipping payload restart.")
```

Comment:
Do we really have to skip it the restart? Why?

Reply:
If installation source is set to the CDN but we are not yet registered, any attempt to interact with the CDN repos will fail. The payload refresh also sets a "Failed to set up installation source" (IIRC) error message even though nothing really is wrong. 

Only after the user has registered is it possible to use the CDN as a source (`redhat.repo` has been generated and the needed authentication tokens are in place) and payload restart has a chance to succeed.

Do you think there is a specific reason to still restart the payload in this case ? In that case we would need some option to skip repo refresh, but I don't see anything like that in the `restart_thread()` method:

https://github.com/rhinstaller/anaconda/blob/be7d7f6cbe96885657351c338254c34b37268a1c/pyanaconda/payload/manager.py#L115